src/components/Grid.re

```diff
@@ -1,32 +1,83 @@
 open Css;
 
-let gridContainer = (spacing: int) => [
-  width(`calc((`add, pct(100.0), px(spacing)))),
-  margin(px(- spacing / 2)),
+type flexBasisLength = [
+  length
+  | `auto
+  | `fill
+  | `content
+  | `maxContent
+  | `minContent
+  | `fitContent
 ];
 
-let gridItem = (spacing: int) => [padding(px(spacing / 2))];
+type gridItemWidth =
+  | Columns(int)
+  | Grow
+  | Auto
+  | Flex(float, int, flexBasisLength);
 
-let component = ReasonReact.statelessComponent("Grid");
+type containerWrap = [ | `nowrap | `wrap | `wrapReverse];
 
 type gridType =
-  | Container
-  | Item;
+  | Container(containerWrap)
+  | Item(gridItemWidth);
 
-[@react.component]
-let make = (~type_: gridType, ~children) => {
-  let getGridStyles = (): list(rule) => {
-    switch (type_) {
-    | Container =>
-      [display(`flex)]
-      @ gridContainer(Theme.spacingPx.small)
-      @ [media(Breakpoints.up(Md), gridContainer(Theme.spacingPx.medium))]
-
-    | Item =>
-      [flexGrow(1.0), flexShrink(1), flexBasis(auto)]
-      @ gridItem(Theme.spacingPx.small)
-      @ [media(Breakpoints.up(Md), gridItem(Theme.spacingPx.medium))]
+module Styles = {
+  let containerSpacing = (spacing: int) => [
+    width(`calc((`add, pct(100.0), px(spacing)))),
+    margin(px(- spacing / 2)),
+  ];
+  let itemSpacing = (spacing: int) => [padding(px(spacing / 2))];
+
+  let calculateColumnsWidth = (columns: int) =>
+    float_of_int(columns) /. 12.0 *. 100.0;
```

Comment:
What is `/.` and `*.`? What does the dot do?

Reply:
It is for `float` types.